File: scala-src/togos/picgrid/file/FileUtil.scala
package togos.picgrid.file

import java.io.File

object FileUtil
{
	def makeParentDirs( f:File ) = {
		val parentFile = f.getParentFile() 
		if( parentFile != null && !parentFile.exists() ) parentFile.mkdirs()
	}
	
	def deltree( f:File ) {
		if( f.isDirectory() ) {
			for( sf <- f.listFiles() ) {
				deltree( sf )
			}
			f.delete()
		} else if( f.exists() ) {
			f.delete()
		}
	}
}

File: java-src/togos/picgrid/file/FileUtil.java
package togos.picgrid.file;

import java.io.File;

public class FileUtil
{
	public static void makeParentDirs( File f ) {
		File parentFile = f.getParentFile(); 
		if( parentFile != null && !parentFile.exists() ) {
			parentFile.mkdirs();
		}
	}
		
	public static void deltree( File f ) {
		if( f.isDirectory() ) {
			File[] list = f.listFiles();
			for( int i=0; i<list.length; ++i ) {
				deltree( list[i] );
			}
		} else if( f.exists() ) {
			f.delete();
		}
	}
	
}

